### ISSUE TITLE
Feature: Emoji Reaction Reinforcement — Learning from 👍 / ❤️ / 👎 on messaging platforms

### ISSUE THREAD

**Issue by danstis** [NousResearch/hermes-agent#27438]
## Problem

Hermes lives on Telegram, Discord, Slack, and other messaging platforms where users naturally react to messages with emoji. Right now those reactions are just… decor. Thumbs up, heart, thumbs down, poo — they’re rich, low-friction feedback signals that go completely unused.

Every time a user reacts to a Hermes message, they’re implicitly saying "good job," "I love this," "that was crap," or "wtf was that." That’s free training data, and it requires zero extra effort from the user — they’re already doing it.

## Proposal

Capture emoji reactions on supported messaging channels and map them into a lightweight reinforcement signal that Hermes can act on.

### Emoji → Signal Mapping

| Reaction | Signal | Weight |
|----------|--------|--------|
| ❤️ heart | Overwhelmingly positive | +2.0 |
| 👍 thumbs up | Positive | +1.0 |
| 😂 laugh | Positive/entertaining | +0.8 |
| 🙌 raised hands | Positive/celebratory | +1.0 |
| 👎 thumbs down | Negative | -1.0 |
| 💩 poo | Strongly negative | -2.0 |
| 😢 cry | Negative/disappointing | -1.5 |
| 😡 angry | Strongly negative | -2.0 |

Configurable per user — some people just react with everything.

### What "Learning" Means Here

This is NOT model fine-tuning (that’s #498’s territory). This is about lightweight, in-process reinforcement:

1. **Memory weighting** — Messages that receive positive reactions get higher retrieval priority in future context lookups. Messages that get negative reactions get deprioritized or flagged for review.

2. **Skill confidence scoring** — If a skill is active when a positive reaction comes in, bump its internal confidence/usage score. Negative reaction → dock it. Over time, Hermes learns which skills actually work vs. which ones look good on paper.

3. **Response style tuning** — Track which tone/length/formats get positive reactions per user and per channel, and bias future responses accordingly (without overriding explicit instructions).

4. **Preference extraction** — "You reacted with ❤️ to my movie recommendations but 👎 to my recipe suggestions" → that’s a user preference signal worth persisting.

### Implementation Considerations

- **Scope:** Reactions are channel-scoped and user-scoped. A reaction in a group chat shouldn’t necessarily affect the same user’s private chat behavior (or maybe it should — make it configurable).
- **Noise handling:** Some users react to everything. Include a decay factor and require a minimum signal threshold before acting.
- **Privacy:** This is per-user, per-instance. Not telemetry. Not shared.
- **Transparency:** Hermes should occasionally acknowledge that it’s learning from reactions ("Noted, I’ll stop suggesting Python when you clearly prefer Go"). Not every time — that’d be annoying — but occasionally.

### Relationship to Existing Work

- **#498 (Conversational RL Personalization):** This is the heavy-duty approach — actual model weight updates. The emoji reaction system could *feed into* #498 as a reward signal source, but it’s independently valuable as a lightweight mechanism that works today.
- **#337 (Skill Evolution):** Reaction feedback could inform which skills get auto-improved vs. deprecated.

### Why This Matters

The best feedback systems are the ones users don’t have to think about. Typing "that was helpful" is work. Tapping ❤️ is muscle memory. If Hermes is going to be "the agent that grows with you," it should listen to the signals users are already sending.

**Comment by alt-glitch** [comment]
Duplicate of #18408 (Telegram emoji reaction learning). This broadens the scope to multi-platform but the core concept — capturing emoji reactions as implicit feedback signals — is the same feature request. PR #23731 is already implementing #18408.

**Comment by konsisumer** [comment]
These issues look like facets of **one underlying problem**: #27438, #34660, #46855.

They were grouped by a semantic pass as different symptoms of one underlying problem in `comp/gateway`. A single coordinated fix may resolve them together — flagging for visibility; no action has been taken on the individual issues.

_Automated root-cause grouping. Reply to correct if these are unrelated._

**Comment by GottZ** [comment]
> *This was generated by AI during triage.*

#### Summary

One PR addresses #27438. #27451 implements the Telegram capture and SQLite storage foundation for emoji feedback, but it does not yet implement the proposed reinforcement consumers or multi-platform capture, and its Telegram integration targets the superseded legacy adapter rather than the current plugin path.

#### Related pull requests

- #27451 `best fix` — (+2485/-1) — partial fix; keep open for salvage: the diff adds configurable emoji mapping, authored-message filtering, add/remove aggregation, durable reaction storage, lifecycle hooks, Telegram handling, tests, and documentation. Consistent with the latest keep_open review on #27451, it is not ready for direct merge because current rich/stream replies bypass authored-ID recording, the integration remains in the legacy Telegram adapter instead of the bundled plugin, and min_signal_threshold/decay_days have no automatic runtime consumers.

#### Suggested consolidation

Merge #27451 after porting reaction capture and authored-message tracking to the current bundled Telegram plugin, covering rich and streaming delivery paths, and either wiring or deferring the unused threshold and decay settings. It is the best available fix and should remain the consolidation target, but the contributor's keep_open review identifies concrete current-main blockers that must be addressed before merge; there are no competing PRs to close as duplicates.

#### Complex graph

```mermaid
flowchart LR
    classDef open fill:#dbeafe,stroke:#1d4ed8,color:#1e3a8a
    classDef merged fill:#dcfce7,stroke:#15803d,color:#14532d
    classDef closed fill:#e5e7eb,stroke:#6b7280,color:#1f2937
    classDef unverified fill:#f3f4f6,stroke:#9ca3af,color:#374151
    classDef best stroke-width:3px,stroke:#b45309
    classDef target stroke-width:3px,stroke:#4338ca
    I27438(["issue #27438 (open)"])
    P27451["PR #27451 (open)"]
    P27451 -->|best fix| I27438
    class I27438 open
    class I27438 target
    class P27451 open
    class P27451 best
    click I27438 "https://github.com/NousResearch/hermes-agent/issues/27438"
    click P27451 "https://github.com/NousResearch/hermes-agent/pull/27451"
```
*Graph: solid arrow = fixes / best fix, dashed arrow = partial or unverified (see edge label); boxed group = PRs duplicating each other; amber border = best fix; indigo border = target; gray node = closed or no verify verdict yet (state tag in the node label).*

_Cross-PR triage: Reviewed 1 pull request and 1 issue in this complex. Each diff was read against this issue; Assessment working set: 107 kB of PR diffs, 13 kB of issue/PR text, 5 kB of discussion (5 comments), 2 verify verdicts. verdicts reflect diff content, not PR titles. Part of an automated triage batch._
